packages/realm/src/Realm.ts

```diff
@@ -261,19 +259,20 @@ export class Realm {
 
   public static defaultPath = Realm.normalizePath("default.realm");
 
+  private static internals = new IterableWeakRefs<binding.Realm>();
+
   /**
    * Clears the state by closing and deleting any Realm in the default directory and logout all users.
    * @private Not a part of the public API: It's primarily used from the library's tests.
    */
   public static clearTestState(): void {
     // Close any realms not already closed
-    for (const weakRealm of RETURNED_REALMS) {
-      const realm = weakRealm.deref();
+    for (const realm of this.internals) {
       if (realm && !realm.isClosed) {
         realm.close();
       }
     }
-    RETURNED_REALMS.clear();
+    this.internals.clear();
```

Comment:
```suggestion
    Realm.internals.clear();
```

While technically correct, using `this` in a static method is a bit weird, unless you are specifically trying to get dynamic binding with subclasses.